### PR TITLE
Sort crates before splitting them into groups (+ some improvements)

### PR DESCRIPTION
**Pull request by joao-paulo-parity**
This is useful so that crates always get routed to a specific group for a given version of the source code, which means that jobs for each group can be reliably retried individually to target a specific crate.

### CODE REVIEW

**Review by alvicsam — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

File: scripts/ci/gitlab/check-each-crate.sh [hunk 1]
```diff
@@ -37,7 +39,9 @@ fi
 group=1
 for ((i=0; i < crates_total; i += crates_per_group)); do
   if [ $group -eq "$target_group" ]; then
-    for crate in "${workspace_crates[@]:$i:$crates_per_group}"; do
+    crates_in_group=("${workspace_crates[@]:$i:$crates_per_group}")
+    printf "crates in the group: ${crates_in_group[*]}" >/dev/null # >/dev/null due to "set -x"
+    for crate in "${crates_in_group[@]}"; do
```
Comment (ggwpez) [inline scripts/ci/gitlab/check-each-crate.sh:44]: Just as note: You only check the default features. There is still the possibility that just some features will fail.

**Review by ggwpez — APPROVED** [review]

File: scripts/ci/gitlab/check-each-crate.sh (hunk 1 above)
Comment (bkchr) [reply scripts/ci/gitlab/check-each-crate.sh:44]: This was already mentioned in the initial pr.

Comment (joao-paulo-parity) [reply scripts/ci/gitlab/check-each-crate.sh:48]: There's a separate ticket for the features: https://github.com/paritytech/substrate/issues/12711

**Review by sergejparity — APPROVED** [review]

**Comment by joao-paulo-parity** [comment]
bot merge

**Pull request merged**